src/libopensc/sm.c

```diff
@@ -158,7 +158,16 @@ sc_sm_single_transmit(struct sc_card *card, struct sc_apdu *apdu)
 
 	/* send APDU flagged as NO_SM */
 	sm_apdu->flags |= SC_APDU_FLAGS_NO_SM;
-	rv = sc_transmit_apdu(card, sm_apdu);
+	/* if token is epass2003, using below code*/
+	switch(card->type){
+		case SC_CARD_TYPE_ENTERSAFE_FTCOS_EPASS2003:
```

Comment:
We double check the code, no need do modify in this layer, we will modify the file apdu.c.
The reason is our epass2003 series support plain APDU and cipher APDU, while in cipher APDU, after receive 6C command, then we will need get data back and re-package APDU with cipher mode, at this time, the set_le API seems only modify the LE, and nothing change of APDU, so we will modify apdu.c to solve it.